File: solutions/csharp/line-up/1/LineUp.cs
public static class LineUp
{
    public static string Format(string name, int number) =>
        $"{name}, you are the {number}{OrdinalSuffixFor(number)} customer we serve today. Thank you!";

    private static string OrdinalSuffixFor(int number) =>
        (number % 100) switch
        {
            11 or 12 or 13 => "th",
            _ => (number % 10) switch
            {
                1 => "st",
                2 => "nd",
                3 => "rd",
                _ => "th"
            }
        };
}

File: solutions/fsharp/line-up/1/LineUp.fs
module LineUp

let private suffix number =
    match number % 100 with
    | 11 | 12 | 13 -> "th"
    | _ ->
        match number % 10 with
        | 1 -> "st"
        | 2 -> "nd"
        | 3 -> "rd"
        | _ -> "th"

let format (name: string) (number: int): string =
    $"{name}, you are the {number}{suffix number} customer we serve today. Thank you!"
